### ISSUE TITLE
Clarification on Package Versions and GRPM α Parameter Setting

### ISSUE THREAD

**Issue by xieyvhuan** [yanyi74/GREP#1]
Hello, thank you for your excellent work. I'm very interested in your research, but I have a couple of questions:

I noticed that your project follows the AA setting, but there seems to be a mismatch between the README.md and the requirements.txt in terms of package dependencies. Which one should I follow for correct environment setup?

Regarding the GRPM module you proposed, the paper mentions an additional parameter α (alpha). Could you kindly clarify how this parameter should be set in practice? Also, would it be possible for you to share an updated version of the run script that includes this setting?

Thank you in advance! @yanyi74

**Comment by xieyvhuan** [comment]
@yanyi74 Hello, thank you for sharing this excellent work. I have been trying to reproduce your experiments and encountered a few issues that I’d like to ask about:

Reproduction Settings
I strictly followed the package versions specified in your requirements.txt (rather than those in the README.md), and noticed that you fixed the α value to 0.2. Under this setting, I ran the following commands on a NVIDIA RTX 4090 GPU:
bash scripts/run_bert.bash bert 0 66
bash scripts/run_bert.bash bert_evi 0.1 66
I also performed the corresponding ISCF inference. The results on the Dev set matched those reported in your paper, or were even slightly better. However, when I submitted the predictions to Codalab, the Test set results were significantly different:
RE_F1: 0.6258  
Evi_F1: 0.0000  
RE_ignore_annotated_F1: 0.6042  
RE_ignore_distant_F1: 0.5640
Have you encountered such a large discrepancy between Dev and Test performance before? What might be the possible causes for this?

Memory Issues with RoBERTa
In your paper, you mentioned that all experiments were conducted on a NVIDIA RTX 3090 GPU. However, when I tried to reproduce the RoBERTa-based DocRED experiment, I encountered Out of Memory (OOM) errors even on a 24GB RTX 4090. Could you please share the GPU memory size you used and your settings for batch_size, gradient_accumulation_steps, etc.? Do I need to reduce the batch size further or switch to FP16 to make RoBERTa run successfully?

Thank you very much for your time and assistance!

**Comment by yanyi74** [comment]
Hello, my code will be re-uploaded shortly. The current version was uploaded last year. The updated version will be available soon, and I will release the specific parameter details at that time!

**Comment by xieyvhuan** [comment]
> 您好，我的代码将很快重新上传。当前版本是去年上传的。更新版本即将上线，具体参数详情届时我会发布！

Thank you for your response!

I truly appreciate your work on this model. I'm currently exploring ways to extend and improve it for my own research, and I would really like to build upon your official implementation.

Would it be possible to share a tentative timeline for the updated code release, or perhaps provide a temporary version of the training script/configs you used to reproduce the reported results (especially with RoBERTa)? Even partial details would be extremely helpful at this stage.

Thank you again for your time and effort!

**Comment by yanyi74** [comment]
As for the significant difference between the dev and test sets you mentioned, I haven't encountered that issue. My guess is that my experiments weren't run on a 4090 GPU, and your package might differ from the version I used previously. As long as the dataset distribution remains consistent, such discrepancies shouldn't occur。And RoBERTa memory issue. I will share the script later and provide these details.

**Comment by yanyi74** [comment]
> > 您好，我的代码将很快重新上传。当前版本是去年上传的。更新版本即将上线，具体参数详情届时我会发布！
> 
> 感谢您的回复！
> 
> 我非常欣赏你在这个模型上的工作。我目前正在探索如何扩展和改进它，用于我自己的研究，并且非常希望能够以你的正式实现为基础。
> 
> 能否分享一下更新代码发布的暂定时间表，或者提供一下你用来复现报告结果（尤其是使用 RoBERTa）的训练脚本/配置的临时版本？在这个阶段，即使是部分细节也会非常有帮助。
> 
> 再次感谢您付出的时间和精力！

 Could you show me your training script? I'm currently outside and don't have access to the lab server to check the script.

**Comment by xieyvhuan** [comment]
Thank you for your reply!

Regarding the dev-test discrepancy, I also initially assumed it could be due to environment differences. However, since the dataset split remains the same and the model is deterministic given a fixed seed, I'm wondering whether differences in GPU architecture (e.g., RTX 4090 vs. 3090) could actually lead to such a large drop in test performance. I would expect such influence to be minimal, but please correct me if you've observed otherwise.

For reference, I strictly followed the `requirements.txt` file to set up the environment, not the `README.md`. All package versions match exactly — including `transformers==4.14.1`, `torch==1.11.0`, etc. — **except for one**: `python-dateutil==2.9.0.post0`. I would like to confirm, **which version of Python was used during your original experiments?** This might help ensure full reproducibility.

Additionally, I would really appreciate it if you could share your GPU memory setup when running the RoBERTa-based model on DocRED. I encountered OOM issues even with gradient accumulation, so I'm curious how much VRAM was available on your 3090 setup, and whether any specific memory-saving techniques were used.

Looking forward to your updated code release and further clarification — I'm working on adapting and extending your model, and your help would be invaluable!

**Comment by xieyvhuan** [comment]
Sure, I’ve attached the training script I’m using below for your reference.

This script is adapted from the `AA` project — I’m currently working on extending it for some analysis. I’ve commented out the `use_graph` argument since I’m not using the graph module in my current experiments. Additionally, I’ve added a new hyperparameter `doc_alpha`, corresponding to the α value mentioned in your paper, to control the weight of the document-level loss during training.

Please see below:

run_bert.bash:
=================================
#!/bin/bash
TYPE=$1
LAMBDA=$2
SEED=$3

NAME=${TYPE}_lambda${LAMBDA}_seed${SEED}
python run.py --do_train \
--data_dir dataset/docred \
--transformer_type bert \
--model_name_or_path bert-base-cased \
--save_path ${NAME} \
--train_file train_annotated.json \
--dev_file dev.json \
--train_batch_size 4 \
--test_batch_size 8 \
--gradient_accumulation_steps 1 \
--num_labels 4 \
--lr_transformer 5e-5 \
--max_grad_norm 1.0 \
--evi_thresh 0.2 \
--evi_lambda ${LAMBDA} \
--doc_alpha 0.1 \
--warmup_ratio 0.06 \
--num_train_epochs 30.0 \
--seed ${SEED} \
--num_class 97
#--use_graph
=================================
iscf_bert.bash：

MODEL_DIR=$1
MODEL_EVI_DIR=$2
SPLIT=$3

python run.py --data_dir dataset/docred \
--transformer_type bert \
--model_name_or_path bert-base-cased \
--load_path ${MODEL_EVI_DIR} \
--eval_mode single \
--test_file ${SPLIT}.json \
--test_batch_size 8 \
--num_labels 4 \
--evi_thresh 0.2 \
--num_class 97
#--use_graph


python run.py --data_dir dataset/docred \
--transformer_type bert \
--model_name_or_path bert-base-cased \
--load_path ${MODEL_DIR} \
--results_path ${MODEL_EVI_DIR} \
--eval_mode fushion \
--test_file ${SPLIT}.json \
--test_batch_size 32 \
--num_labels 4 \
--evi_thresh 0.2 \
--num_class 97
#--use_graph
=================================
run_roberta.bash：

#!/bin/bash
TYPE=$1
LAMBDA=$2
SEED=$3

NAME=${TYPE}_lambda${LAMBDA}_seed${SEED}
python run.py --do_train \
--data_dir dataset/docred \
--transformer_type roberta \
--model_name_or_path roberta-large \
--train_file train_annotated.json \
--dev_file dev.json \
--save_path ${NAME} \
--train_batch_size 2 \
--test_batch_size 8 \
--gradient_accumulation_steps 1 \
--num_labels 4 \
--lr_transformer 3e-5 \
--max_grad_norm 1.0 \
--evi_thresh 0.2 \
--evi_lambda ${LAMBDA} \
--doc_alpha 0.1 \
--warmup_ratio 0.06 \
--num_train_epochs 30.0 \
--seed ${SEED} \
--num_class 97
#--attn_heads 1 \
#--gcn_layers 1 \
#--iters 2
#--use_graph

**Comment by xieyvhuan** [comment]
Sure, I’ve attached the training script I’m using below for your reference.

This script is adapted from the project — I’m currently working on extending it for some analysis. I’ve commented out the argument since I’m not using the graph module in my current experiments. Additionally, I’ve added a new hyperparameter , corresponding to the α value mentioned in your paper, to control the weight of the document-level loss during training.AAuse_graphdoc_alpha

Please see below:

run_bert.bash:
===============================
#!/bin/bash
TYPE=$1
LAMBDA=$2
SEED=$3

NAME=${TYPE}_lambda${LAMBDA}_seed${SEED}
python run.py --do_train
--data_dir dataset/docred
--transformer_type bert
--model_name_or_path bert-base-cased
--save_path ${NAME}
--train_file train_annotated.json
--dev_file dev.json
--train_batch_size 4
--test_batch_size 8
--gradient_accumulation_steps 1
--num_labels 4
--lr_transformer 5e-5
--max_grad_norm 1.0
--evi_thresh 0.2
--evi_lambda ${LAMBDA}
--doc_alpha 0.1
--warmup_ratio 0.06
--num_train_epochs 30.0
--seed ${SEED}
--num_class 97
#--use_graph

iscf_bert.bash：
===============================
MODEL_DIR=$1
MODEL_EVI_DIR=$2
SPLIT=$3

python run.py --data_dir dataset/docred
--transformer_type bert
--model_name_or_path bert-base-cased
--load_path ${MODEL_EVI_DIR}
--eval_mode single
--test_file ${SPLIT}.json
--test_batch_size 8
--num_labels 4
--evi_thresh 0.2
--num_class 97
#--use_graph

python run.py --data_dir dataset/docred
--transformer_type bert
--model_name_or_path bert-base-cased
--load_path ${MODEL_DIR}
--results_path ${MODEL_EVI_DIR}
--eval_mode fushion
--test_file ${SPLIT}.json
--test_batch_size 32
--num_labels 4
--evi_thresh 0.2
--num_class 97
#--use_graph

run_roberta.bash：
===============================
#!/bin/bash
TYPE=$1
LAMBDA=$2
SEED=$3

NAME=${TYPE}_lambda${LAMBDA}_seed${SEED}
python run.py --do_train
--data_dir dataset/docred
--transformer_type roberta
--model_name_or_path roberta-large
--train_file train_annotated.json
--dev_file dev.json
--save_path ${NAME}
--train_batch_size 2
--test_batch_size 8
--gradient_accumulation_steps 1
--num_labels 4
--lr_transformer 3e-5
--max_grad_norm 1.0
--evi_thresh 0.2
--evi_lambda ${LAMBDA}
--doc_alpha 0.1
--warmup_ratio 0.06
--num_train_epochs 30.0
--seed ${SEED}
--num_class 97
#--attn_heads 1
#--gcn_layers 1
#--iters 2
#--use_graph

**Comment by xieyvhuan** [comment]
To successfully run the RoBERTa model, I reduced the **train_batch_size** from 4 to 2.

**Comment by xieyvhuan** [comment]
I also thought of another possible reason. Around line 315 of the prepro.py file, there's a line save_graphs(graph_list, "your_path"). I commented it out because I wasn't sure about its purpose or the required save path. At the time, I assumed it might have been intended for your later debugging or post-analysis of the graphs. @yanyi74

**Comment by xieyvhuan** [comment]
Hello,

I encountered an issue while training an unsupervised RoBERTa model and would appreciate your insights on it.

I ran the following command:

”bash scripts/run_roberta.bash roberta 0 66“

This script is intended to train an unsupervised RoBERTa model. However, during training, I noticed that the evaluation metrics (dev_rel, dev_rel_ign, dev_evi) dropped to zero, and I repeatedly saw warnings like the following in the logs:

UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 due to no predicted samples.

An example log output is:

{'dev_rel': [0.0, 0.0, 0], 'dev_rel_ign': [0.0, 0.0, 0], 'dev_evi': [0.0, 0.0, 0]}

This issue only occurs in the unsupervised setting. When I use the same script for supervised RoBERTa training, the metrics and predictions are normal.

I should note that I’ve seen this warning before in other training processes, but I assumed it was related to global document-level relation prediction and didn’t affect overall performance — since training continued normally in those cases, I ignored it. However, this time the metrics remain zero, so it seems to indicate a more critical issue.

Here are some possible causes I suspect:

The model might be outputting no predictions during inference in the unsupervised setup;

After certain training steps (e.g., learning rate decay), the model might lose its discriminative capacity;

Some configuration or processing may be missing or incompatible in the unsupervised pipeline (e.g., labels not used or outputs not matching expected format);

My own code modifications (e.g., adding doc_alpha, disabling use_graph) may have introduced side effects.
